socorro/unittest/cron/jobs/test_update_signatures.py

```diff
@@ -0,0 +1,257 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import mock
+
+from socorro.cron.crontabber_app import CronTabberApp
+from socorro.unittest.cron.jobs.base import IntegrationTestBase
+
+
+class FakeModel(object):
+    def __init__(self):
+        self._get_steps = []
+
+    def add_get_step(self, assertions, response):
+        assertions = assertions or []
```

Comment:
These assertions aren't used anywhere in the tests, should we drop them?

Actually, without the support for assertions, we can achieve what this is doing by assigning an iterable to the mock SuperSearch. Each time it's called, it will return the next item in the iterable, and raise an error if it is called more times than there are values:

```python
@mock.patch('socorro.cron.jobs.update_signatures.SuperSearch')
def test_no_crashes_to_process(self, mock_supersearch):
    mock_supsersearch.side_effect = [
        {
            'errors': [],
            'hits': [],
            'total': 0,
            'facets': {}
        },
    ]
```

Reply:
Yeah! Except that this is a `FakeModel` and I'm thinking of uplifting it and using it for other things, too, since it's generalized to the middleware models. The assertions bit lets us verify the parameters passed in.

So, we could go one of two ways:

1. fuck the future and reduce it now
2. write up a bug to uplift it and use it for mocking other models